src/main/java/frc/robot/command/AutoAlignCommand.java

```diff
@@ -0,0 +1,217 @@
+  // These offsets are translational adjustments (in meters) relative to the master node.
+  private final Map<AutoAlignMode, Translation2d> subNodeOffsets = new HashMap<>() {{
+      put(AutoAlignMode.CORAL_LEFT, new Translation2d(-0.3, 0.0));
+      put(AutoAlignMode.CORAL_RIGHT, new Translation2d(0.3, 0.0));
+      put(AutoAlignMode.ALGAE_MIDDLE, new Translation2d(0.0, 0.3));
+  }};
+
+  // Trajectory configuration taken from settings.json values (adjust as necessary)
+  private final TrajectoryConfig trajConfig = new TrajectoryConfig(3.0, 3.0)
+          .setKinematics(drive.kinematics);
+
+  public AutoAlignCommand(DriveSubsystem drive, PositioningSubsystem positioning, ControllerInput controller) {
+    this.drive = drive;
+    this.positioning = positioning;
+    this.controller = controller;
+    addRequirements(drive);
+  }
+
+  @Override
+  public void initialize() {
+    // Determine which auto-align button is pressed; assume only one is held.
+    if (controller.getAutoAlignCoralLeft()) {
+      selectedMode = AutoAlignMode.CORAL_LEFT;
+    } else if (controller.getAutoAlignCoralRight()) {
+      selectedMode = AutoAlignMode.CORAL_RIGHT;
+    } else if (controller.getAutoAlignAlgaeMiddle()) {
+      selectedMode = AutoAlignMode.ALGAE_MIDDLE;
+    } else {
+      // No button is held; cancel command.
+      cancel();
+      return;
+    }
+
+    // Get current robot pose from PositioningSubsystem
+    Pose2d currentPose = positioning.getRobotPose();
+
+    // Find the closest master node using Euclidean distance
+    Pose2d closestMaster = null;
+    double minDistance = Double.POSITIVE_INFINITY;
+    for (Pose2d master : masterNodes) {
+      double distance = currentPose.getTranslation().getDistance(master.getTranslation());
+      if (distance < minDistance) {
+        minDistance = distance;
+        closestMaster = master;
+      }
+    }
+
+    if (closestMaster == null) {
+      // Safety: if no master node is found, cancel command.
+      cancel();
+      return;
+    }
+
+    // Generate the absolute trajectory from the current pose to the closest master node.
+    absoluteTrajectory = PathPlanner.generateTrajectory(
+            currentPose,
+            List.of(), // no intermediate waypoints
+            closestMaster,
+            trajConfig
+    );
+
+    // Determine the desired sub node offset based on the selected mode.
+    Translation2d subOffset = subNodeOffsets.get(selectedMode);
+    // Compute final target for the second phase:
+    Pose2d subTarget = new Pose2d(closestMaster.getTranslation().plus(subOffset), closestMaster.getRotation());
+
+    // Generate a relative trajectory from the master node to the sub node.
+    // This trajectory is only translational; the rotation remains as the master node's rotation.
+    relativeTrajectory = PathPlanner.generateTrajectory(
+            closestMaster,
+            List.of(), // no intermediate waypoints
+            subTarget,
+            trajConfig
+    );
```

Comment:
This is a hallucination. There's no such thing as `PathPlanner.generateTrajectory`

Refer to the [PathPlanner docs](https://pathplanner.dev/pplib-create-a-path-on-the-fly.html) for how to actually do this.